sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -388,7 +388,9 @@ def test_cos():
 
     assert cos(k*pi) == (-1)**k
     assert cos(2*k*pi) == 1
-
+    assert cos(0, evaluate=False).simplify() == 1
+    assert cos(Rational(1, 2)).is_zero is False
+    assert cos(asin(-1, evaluate=False), evaluate=False).simplify() == 0
```

Comment:
These should test `is_zero` rather than `simplify` although it's fine to also test `simplify`.

Reply:
The following error shows up for line 393
`assert cos(asin(-1, evaluate=False), evaluate=False).is_zero is True`
AssertionError

although this test passes
`assert cos(asin(-1, evaluate=True), evaluate=True).is_zero is True`

Reply:
@oscarbenjamin please have a look

Reply:
That's because `is_zero` gives None which is fine because it's always logically correct to return None in three valued logic. Just assert that the result is None but note in a comment that really it should be True even if it is not always possible to resolve an assumptions query.